### ISSUE TITLE
Parsing tables with missing `</td>` *and* `</tr>` is different than in browsers

### ISSUE THREAD

**Issue by sylbru** [danneu/elm-html-parser#5]
Hi, and thank you for this package that will probably allow me to get a proof-of-concept out (and maybe more) much more easily!

I’m having an issue with really bad HTML coming from the backend, where the result when parsed with the library is different than what the browsers do.

Basically it’s a table with missing `</td>` and `</tr>` at the end of each row:

```
<table>
<tr>
    <td>A1</td>
    <td>B1
<tr>
    <td>A2</td>
    <td>B2
<tr>
    <td>A3</td>
    <td>B3
</table>
```

Instead of returning the correct 2×3 table (as if the missing ending tags were there), the result is that the second row is nested into the second cell of the first row, the third row is nested into the second cell of that nested row, etc.

Expected:
![image](https://user-images.githubusercontent.com/3757523/209413830-b2ddef96-e729-4d07-b8d9-5c084e90dfc7.png)

Actual:
![image](https://user-images.githubusercontent.com/3757523/209413817-4493f39c-d248-4a96-9090-9447ee83c21f.png)

In my case, I will probably fix the backend since I can. But because your goal seems to be to match the behaviour of browsers, the issue seemed relevant to mention here.

**Comment by danneu** [comment]
Good find! 

At a glance, I suppose the `element` parser can be patched to special-case more autoclosing rules.

Thinking out loud, since I'm not on a dev machine at the moment:

Since `tr` and `td` are in the parser's list of autoclosing tags, they already automatically close when encountering an open tag with their same name, like `<tr>a<tr>b`. 

I think, the minimal fix for this issue is to make open `<tr>` autoclose `<td>` like how I special-case open `<body>` to autoclose `<head>`:

https://github.com/danneu/html-parser/blob/cfa2acb7e44c6f675b9c30019ae4ec71a4e6dfd4/src/Html/Parser.elm#L525-L540

Maybe this is sufficient:

```diff
  if tag == "head" then 
      notNode cfg [ tag, "body" ] 
+ else if tag == "td" then
+    notNode cfg [ tag, "tr" ] 
  else 
      ...
```

Thanks for your time. I'll have a look.

**Comment by danneu** [comment]
I published this fix as v2.0.3: https://package.elm-lang.org/packages/danneu/html-parser/

**Comment by sylbru** [comment]
Works perfectly, thank you so much!
